pandas/core/indexing.py

```diff
@@ -1091,8 +1095,17 @@ def _convert_to_indexer(self, obj, axis=0, is_setter=False):
         """
         labels = self.obj._get_axis(axis)
 
-        # if we are a scalar indexer and not type correct raise
-        obj = self._convert_scalar_indexer(obj, axis)
+        if isinstance(obj, slice):
+            return self._convert_slice_indexer(obj, axis)
+
+        # try to find out correct indexer, if not type correct raise
+        try:
+            obj = self._convert_scalar_indexer(obj, axis)
+        except TypeError:
+
+            # but we will allow setting
+            if is_setter:
```

Comment:
So I was fixing this bug

```
In [1]: s = Series([1,2,3])

In [2]: s
Out[2]: 
0    1
1    2
2    3
dtype: int64

In [3]: s.index
Out[3]: RangeIndex(start=0, stop=3, step=1)

In [4]: s['0'] = 5

In [5]: s
Out[5]: 
0    1
1    2
2    3
0    5
dtype: int64

In [6]: s.index
Out[6]: Index([0, 1, 2, u'0'], dtype='object')
```

so ``s[1.0] = 1`` should change this to a ``Float64Index``

but I couldn't get it to work and it seemed innocuous enough. Let me have another look. This is a giant rabbit hole.